### PR TITLE
Add option to verify written flash

### LINKED ISSUE #236: Add option to verify flash
It would be nice if you could add a cli option to verify the written flash, as this is now possible with probe-rs 0.11.0, but it default to false.

See https://github.com/probe-rs/probe-rs/pull/671/files for more details.

### PR DESCRIPTION
**Pull request by Suyash458**
- add option to verify written flash
 - fixes #236

### CODE REVIEW

**Comment by Suyash458** [comment]
@Urhengulas I've got an ST Link JTAG programmer and an ESP32, would love some tips on testing this out!

File: src/cli.rs [hunk 1]
```diff
@@ -85,6 +85,10 @@ pub struct Opts {
     /// Arguments passed after the ELF file path are discarded
     #[structopt(name = "REST")]
     _rest: Vec<String>,
+
+    /// Verifies the written program.
+    #[structopt(long)]
+    pub verify: bool,
```
Comment (Urhengulas) [inline src/cli.rs:91]: Please move this before `_rest`.

**Review by Urhengulas — CHANGES_REQUESTED** [review]
> @Urhengulas I've got an ST Link JTAG programmer and an ESP32, would love some tips on testing this out!

Sure!

1. Install your local `probe-run`
    ```console
    $ cd path/to/probe-run
    $ cargo install --path .
    ```
2. Set up the https://github.com/knurling-rs/app-template for your board
3. Add the `--verify` option to the `runner` in `.cargo/config.toml` ([link](https://github.com/knurling-rs/app-template/blob/main/.cargo/config.toml#L3))
4. Then just run a few of the examples, e.g.:
    ```console
    $ cargo rb hello
    ```
5. Try it in combination with some of the other configuration flags, e.g. `--measure-stack`, `--no-flash`, `--connect-under-reset`
6. If that behaves as expected you can mark the PR as ready for review.

PS: Please add the change to the `CHANGELOG.md`.

**Comment by Suyash458** [comment]
> > @Urhengulas I've got an ST Link JTAG programmer and an ESP32, would love some tips on testing this out!
> 
> Sure!
> 
> 1. Install your local `probe-run`
>    ```
>    $ cd path/to/probe-run
>    $ cargo install --path .
>    ```
> 2. Set up the https://github.com/knurling-rs/app-template for your board
> 3. Add the `--verify` option to the `runner` in `.cargo/config.toml` ([link](https://github.com/knurling-rs/app-template/blob/main/.cargo/config.toml#L3))
> 4. Then just run a few of the examples, e.g.:
>    ```
>    $ cargo rb hello
>    ```
> 5. Try it in combination with some of the other configuration flags, e.g. `--measure-stack`, `--no-flash`, `--connect-under-reset`
> 6. If that behaves as expected you can mark the PR as ready for review.
> 
> PS: Please add the change to the `CHANGELOG.md`.

@Urhengulas Turns out the ST Link debuggers don't work with the ESP32 
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/index.html#jtag-debugging-selecting-jtag-adapter

Would it be possible to test this out on your end?

**Review by Urhengulas — CHANGES_REQUESTED** [review]
I tested it and it seems to verify it. At least the `probe-rs` logs say so.

Looking into the code I realized that the `--verify` option would be skipped, if used together with `--no-flash`. Can you therefore please change the attribute of `no_flash` (not `verify`!) to following:
```diff
/// Skip writing the application binary to flash.
-    #[structopt(long, conflicts_with = "defmt")]
+    #[structopt(
+        long,
+        conflicts_with = "disable-double-buffering",
+        conflicts_with = "verify"
+    )]
    pub no_flash: bool,
```
(`defmt` isn't needed anymore, and `disable-double-buffering` has no effect with `--no-flash` as well)

Other than that it looks good to me!

**Review by Urhengulas — CHANGES_REQUESTED** [review]
Can you please run `cargo fmt`?
Afterwards you can use `bors r+` to merge.

bors d+

**Pull request merged**